### ISSUE TITLE
Time freezing

### ISSUE THREAD

**Issue by george-martinec** [marcinjakubowski/date-menu-formatter#14]
Time sometimes freezes and you need to turn off and turn of the extension

Ubuntu 22.04.2 LTS

**Comment by kimjetwav** [comment]
I just had a similar issue. Debian Unstable: time froze at 11:40:45 and only unfroze after  I restarted the extension. First time I've seen this.

**Comment by thiagobraga** [comment]
Is there a way to get logs from this extension?

I'm having the same problem and it's very annoying. So I followed `syslog` with `sudo tail -f /var/log/syslog -n 1000 | grep gnome-shell` since yesterday (or with `journalctl` for example), but nothing appears searching for `date-menu-formatter` or `formatter`.

Anyway, thanks for the extension :)

**Comment by marcinjakubowski** [comment]
Not really, there's not much logging going on. I wonder how that's possible
though, time is updated via setTimeout every second, I have never seen it
fail on any of my boxes. Maybe there's a more GTK way of setting up a
timer, will need to dig into that.

sob., 3 cze 2023, 15:17 użytkownik Thiago Braga ***@***.***>
napisał:

**Comment by thiagobraga** [comment]
I get some information, maybe it can help. Today it freezed at `11:00:32` Brazilian time, HHmmss. So I tried to find something around this moment in `syslog`. I did `sudo cat /var/log/syslog | grep "Aug  3 11:00:3"`:
```
Aug  3 11:00:30 thiago-notebook avahi-daemon[906]: Joining mDNS multicast group on interface veth3f5060f.IPv6 with address fe80::40f8:2eff:fe65:2843.
Aug  3 11:00:30 thiago-notebook avahi-daemon[906]: New relevant interface veth3f5060f.IPv6 for mDNS.
Aug  3 11:00:30 thiago-notebook avahi-daemon[906]: Registering new address record for fe80::40f8:2eff:fe65:2843 on veth3f5060f.*.
Aug  3 11:00:30 thiago-notebook chrome-fjbkmdmgdpajpcpndphdhnfipaemojoi-Profile_1.desktop[153909]: Fontconfig error: Cannot load default config file: No such file: (null)
Aug  3 11:00:33 thiago-notebook gnome-shell[4184]: Attempting to call back into JSAPI during the sweeping phase of GC. This is most likely caused by not destroying a Clutter actor or Gtk+ widget with ::destroy signals connected, but can also be caused by using the destroy(), dispose(), or remove() vfuncs. Because it would crash the application, it has been blocked and the JS callback not invoked.
Aug  3 11:00:33 thiago-notebook gnome-shell[4184]: Attempting to call back into JSAPI during the sweeping phase of GC. This is most likely caused by not destroying a Clutter actor or Gtk+ widget with ::destroy signals connected, but can also be caused by using the destroy(), dispose(), or remove() vfuncs. Because it would crash the application, it has been blocked and the JS callback not invoked.
Aug  3 11:00:33 thiago-notebook gnome-shell[4184]: == Stack trace for context 0x55d632974170 ==
Aug  3 11:00:33 thiago-notebook gnome-shell[4184]: == Stack trace for context 0x55d632974170 ==
Aug  3 11:00:36 thiago-notebook systemd[1]: NetworkManager-dispatcher.service: Deactivated successfully.
```
<img src="https://github.com/marcinjakubowski/date-menu-formatter/assets/993684/be620c77-51f1-487f-a955-7eb83b5b260d" width="400" />
This warning specifically. Does it have something related with Date Menu Formatter?
```
Aug  3 11:00:33 thiago-notebook gnome-shell[4184]: Attempting to call back into JSAPI during the sweeping phase of GC. This is most likely caused by not destroying a Clutter actor or Gtk+ widget with ::destroy signals connected, but can also be caused by using the destroy(), dispose(), or remove() vfuncs. Because it would crash the application, it has been blocked and the JS callback not invoked.
```

**Comment by richbl** [comment]
Running Ubuntu 23.10, and I see this extension freeze my time only occasionally when I bring my Lenovo T15 out of hibernation (both running Wayland and X11).

It's happened several times, and it actually quite unsettling as--evidently--I rely on that clock more than I thought I did, and when time grinds to a halt (particularly in a heinous meeting)... I'm mortified.

**Comment by Dr-Axen** [comment]
I have the same issue. I am on Ubuntu 22.04.4 LTS. I do not use suspend or hibernate. Sometimes my desktop becomes unresponsive for a minute, but I am not sure if this is related to the time freezing issue.

**Comment by bomdia** [comment]
I'm not very sure of what happens, but to me seems a problem with gnome itself as we currently use the method that the official gjs guide for writing extension and application say to use, it's very strange it happened sometime to me too but no logging and no error, just stopped working, we can't do more that what we already do, to me the problem lie in some thing that the shell do that invalidate the timer (a glib timeout) not something we can do anything about.

**Comment by bomdia** [comment]
@JElchison it's month to me too that it doesn't freeze, but saying that it's not the fault of the extension its about 99% of probability, as we use the correct method to do the timer and it is an issue experienced across multiple extensions this git repo isn't the correct place to open the issue, should be opened on glib repo or gnome shell repo, the only time that happened to me i resolved changing the time of the refresh (changing it destroy the old timer and create a new one) this is my motivation for closing it, maybe i'm wrong but the last time that happened i was on gnome 46, with gnome 47 on fedora never happened again, if you want to do some test you are welcome, but in any case from the part of the extension we can't do anything more
